homework/index.js

```diff
@@ -29,25 +32,148 @@
     return elem;
   }
 
-  function renderRepoDetails(repo, ul) {
-    createAndAppend('li', ul, { text: repo.name });
+  // 1- create function to add row
+  function createRow(table, label, content) {
+    const row = createAndAppend('tr', table);
+    createAndAppend('th', row, { text: label });
+    createAndAppend('td', row, { text: content });
+    return row;
+  }
+
+  // 2- create details of repositories
+  function renderRepoDetails(detailsSection, repo) {
+    const li = createAndAppend('li', detailsSection, { class: 'item' });
+    const table = createAndAppend('table', li);
+    // 2-1 add repository name
+    const repoName = createRow(table, 'Repository:', '');
+    createAndAppend('a', repoName.lastChild, {
+      href: repo.html_url,
+      text: repo.name,
+      target: '_blank',
+    });
+    // 2-2 add description
+    createRow(table, 'Description:', repo.description || 'N/A');
+    // 2-3 add Forks
+    createRow(table, 'Forks:', repo.forks);
+    // 2-4 add updated
+    createRow(table, 'Updated:', new Date(repo.updated_at).toLocaleString());
+  }
+
+  // 3- Create function to sort the repositories
+  function sorting(part1, part2) {
+    return part1.name.localeCompare(part2.name);
+  }
+
+  // 4- Creating the contributions part
+  function renderContributor(contributor, ul) {
+    const contributorItem = createAndAppend('li', ul, {
+      class: 'contributor-item',
+    });
+    const contributorLink = createAndAppend('a', contributorItem, {
+      href: contributor.html_url,
+      target: '_blank',
+    });
+    const contributorDiv = createAndAppend('div', contributorLink, {
+      class: 'contributor',
+    });
+    createAndAppend('img', contributorDiv, {
+      src: contributor.avatar_url,
+      class: 'contributor-avatar',
+    });
+    const contributorDetails = createAndAppend('div', contributorDiv, {
+      class: 'contributor-details',
+    });
+    createAndAppend('div', contributorDetails, {
+      text: contributor.login,
+      class: 'contributor-login',
+    });
+    createAndAppend('div', contributorDetails, {
+      text: contributor.contributions,
+      class: 'contributions-number',
+    });
+  }
+
+  function createContributors(contributorSection, url) {
+    createAndAppend('h4', contributorSection, {
+      text: 'Contributions:',
+    });
+    fetchJSON(url)
+      .then(contributors => {
+        if (contributors && contributors.length) {
+          const ul = createAndAppend('ul', contributorSection);
+          contributors.forEach(contributor =>
+            renderContributor(contributor, ul),
+          );
+        } else {
+          createAndAppend('div', contributorSection, {
+            text: 'N/A',
+            class: 'not-available',
+          });
+        }
+      })
+      .catch(err =>
+        createAndAppend('div', contributorSection, {
+          text: err.message,
```

Comment:
Where did the `alert-error` class go?

Reply:
HI @remarcmij, I forgot to add after I applied the modification. now it is there again. 

Thanks  :+1: